wails-frontend/src/pages/backupKeyPairs.tsx

```diff
@@ -0,0 +1,100 @@
+import { useState } from 'react';
+import { promptRequest } from '../events/events';
+import { useLocation } from 'react-router-dom';
+import { handleCancel } from '../utils/utils';
+import { Password } from '@massalabs/react-ui-kit/src/components/Password/Password';
+import { Button } from '@massalabs/react-ui-kit/src/components/Button/Button';
+
+const BackupKeyPairs = () => {
+  const [input, setInput] = useState<string | undefined>(undefined);
+  const [show, setShow] = useState<boolean>(false);
+  const [privateKey, setPkey] = useState<string | undefined>(undefined);
+
+  let { state } = useLocation();
+  const req: promptRequest = state.req;
+
+  const baselineStr = `Enter your password to show your private key`;
+
+  const handleCopy = () => {
+    console.log('copy');
+  };
+
+  const handleShow = () => {
+    if (!input || !input.length) {
+      return;
+    }
+
+    if (!privateKey) {
+      // TODO: validate private key
+      setPkey(input);
+      setInput('');
+      setShow(!show);
+      return;
+    }
+  };
+
+  const updateInput = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setInput(e.target.value);
+
+  return (
+    <div className="bg-primary flex flex-col justify-center items-center h-screen w-full">
+      <div className="w-1/4 max-w-sm min-w-fit">
+        {!show ? (
```

Comment:
you are trying to implement to screens in one component. Try to split to two component or separate the two parts in functions

```ts
  const firstPart = () => {
    return (
      <>
        <div>
          <p className="mas-title text-neutral pb-4">{req.Msg}</p>
        </div>
        <div>
          <p className="mas-body text-neutral pb-4">{baselineStr}</p>
        </div>
        <div id="input" className="pb-4">
          <Password
            id="input"
            onChange={updateInput}
            autoComplete="off"
            name="input"
            placeholder={'Password'}
            value={input}
          />
        </div>
        <div className="flex flex-row gap-4 w-full pb-4">
          <div className="min-w-fit w-full">
            <Button variant={'secondary'} onClick={handleCancel}>
              Cancel
            </Button>
          </div>
          <div className="min-w-fit w-full">
            <Button onClick={handleShow}>Show</Button>
          </div>
        </div>
      </>
    );
  };
```

```suggestion
{!show ? firstPart() : secondPart()}
```

Reply:
@mazmassa do you think it's better to create two different components?

Reply:
I would agree with you, it's seems like the most elegant solution

Reply:
changed